amcl/src/amcl_node.cpp

```diff
@@ -1020,6 +1028,97 @@ AmclNode::globalLocalizationCallback(std_srvs::Empty::Request& req,
   return true;
 }
 
+/*
+ * 
+ * name: customPoseGenerator
+ * Sample from a custom set of particles equal to the Max Number of particles
+ * @gemetry_msgs::PoseArray
+ * @pf_vector_t
+ * 
+ */
+
+pf_vector_t AmclNode::customPoseGenerator(void* arg)
+{
+  geometry_msgs::PoseArray* parray = (geometry_msgs::PoseArray*) arg;
+  geometry_msgs::Pose le = parray->poses.back();
+  tfScalar roll, pitch, yaw ;
+  
+  tf::Quaternion quat(le.orientation.x,le.orientation.y,le.orientation.z,le.orientation.w);
+  tf::Matrix3x3(quat).getRPY(roll, pitch, yaw);
+  
+  //std::cout<<"Size of array "<<parray->poses.size()<<std::endl;
+  
+  parray->poses.pop_back();
+  
+  pf_vector_t p;
+  p.v[0] = le.position.x;
+  p.v[1] = le.position.y;
+  p.v[2] = (double)yaw;//drand48() * 2 * M_PI - M_PI;
+  
+  return p;
+}
+
+
+bool AmclNode::setParticlesCallback(amcl::amcl_particles::Request& req,
+                                    amcl::amcl_particles::Response& res)
+{  
+  int no_of_particles =  req.pose_array_msg.poses.size();
+  ROS_INFO("Received set particles srv call, Pose Array Header seq = %i",req.pose_array_msg.header.seq);
+  ROS_INFO("Received set particles srv call, Pose Array Header stamp = %0.4f",req.pose_array_msg.header.stamp.toSec());
```

Comment:
All this output feels a little spammy (7 ROS_INFOs per call). I'd suggest converting all but one of them to ROS_DEBUG_NAMED() calls -- with a good namespace so it's easy to turn on.